docs/console.md

```diff
@@ -0,0 +1,57 @@
+# Console
+
+In addition to the admin dashboard, Flarum provides several console commands to manage your forum.
+
+To use the console:
+
+1. `ssh` into the server where your flarum installation is hosted
+2. `cd` to the folder that contains a file called `flarum`
+3. Run the command via `php flarum [command]`
+
+## Default Commands
+
+### list
+
+Lists all available management commands, as well as instructions for using management commands
+
+### help
+
+`php flarum help [command_name]`
+
+Displays help output for a given command.
+
+You can also output the help in other formats by using the --format option:
+
+`php flarum help --format=xml list`
+
+To display the list of available commands, please use the list command.
+
+### info
+
+`php flarum info`
+
+Gather information about Flarum's core and installed extensions. This is very useful for debugging issues, and should be shared when requesting support.
+
+### cache:clear
+
+`php flarum cache:clear`
+
+Clears the backend flarum cache, including generated js/css, text formatter cache, and cached translations. This should be run after installing or removing extensions, and running this should be the first step when issues occur.
+
+### migrate
+
+`php flarum migrate`
+
+Runs all outstanding migrations. This should be used when an extension that modifies the database is added or updated.
+
+### migrate:reset
+
+`php flarum migrate:reset --extension [extension_id]`
+
+Reset all migrations for an extension. This is mostly used by extension developers, but on occasion, you might need to run this if you are removing an extension, and want to clear all of its data from the database. Please note that the extension in question must currently be installed (but not necessarily enabled) for this to work.
+
+### generate:migration
+
+`php flarum generate:migration --extension [composer_extension_name] [name]`
+
+Generates an empty migration file with a given name for a given extension. This is mostly useful for extension developers. Please note that you should provide the composer extension name instead of the composer id (e.g. `flarum/flags` instead of `flarum-flags`).
```

Comment:
@flarum/core this command is lacking I think. Currently the generate:migration command will save the database file to the vendor extension migrations directory. Not a single extension developer I know of uses the vendor directory for development. For now we should not document this command, maybe we need to drop the command as well.

Reply:
Well, if it's symlinked, shouldn't it get created in the right place?

Reply:
Who says the extension is already installed? But yeah, okay, if it is using symlinks should work.

If you ask me, I'd rather create a developer extension/package (similar to what @datitisev created) which we can ship for developers as a companion toolkit. This kit would not target the vendor directory but the configured workbench directory.

So, question remains. Do we want to keep this docs entry, this is the only developer relating command we have.

Reply:
I'd be fine removing it tbh. I'd also like to add in a little blurb about scheduling commands via fof console.

Reply:
fof console isn't a core feature though so I don't think it makes sense to include it.

Reply:
Fair, although we do document some unofficial utils occasionally. Maybe just a quick info thing? (not a full docs section). And I did open an issue for discussion of maybe making that a core feature: https://github.com/flarum/core/issues/2281